File: CultureLinkCRM/CultureLinkCRM.Web/Modules/ServerTypes/CrmDB/JurisdictionColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { JurisdictionRow } from "./JurisdictionRow";

export interface JurisdictionColumns {
    JurisdictionId: Column<JurisdictionRow>;
    JurisdictionTypeSystemCode: Column<JurisdictionRow>;
    Name: Column<JurisdictionRow>;
    Abbreviation: Column<JurisdictionRow>;
    ListInd: Column<JurisdictionRow>;
}

export class JurisdictionColumns extends ColumnsBase<JurisdictionRow> {
    static readonly columnsKey = 'CrmDB.Jurisdiction';
    static readonly Fields = fieldsProxy<JurisdictionColumns>();
}

File: CultureLinkCRM/CultureLinkCRM.Web/Modules/CrmDB/Jurisdiction/JurisdictionColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace CultureLinkCRM.CrmDB.Columns;

[ColumnsScript("CrmDB.Jurisdiction")]
[BasedOnRow(typeof(JurisdictionRow), CheckNames = true)]
public class JurisdictionColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int JurisdictionId { get; set; }
    public string JurisdictionTypeSystemCode { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Abbreviation { get; set; }
    public int ListInd { get; set; }
}